src/flag_gems/utils/shape_utils.py

```diff
@@ -137,3 +139,30 @@ def dim_compress(inp, dims):
     sorted_reduction_dim = sorted(dims, key=lambda x: stride[x], reverse=True)
     order = batch_dim + sorted_reduction_dim
     return inp.permute(order).contiguous()
+
+
+def offsetCalculator(inp, idx, strides, dim, isInp):
+    ndim = inp.ndim
+    shape = list(inp.shape)
+    offsets = 0
+    idx_dim = 0
+    for d in range(0, ndim):
+        mod = floor(idx % shape[d])
```

Comment:
I think elements of idx and shape should be integers. Is it necessary to use torch.floor?

Reply:
I tested it out, and removing torch.floor definitely doesn't affect the correctness. I've gone ahead and removed it.